web-ui/src/pages/FeedbackSubmitPage.jsx

```diff
@@ -59,13 +71,31 @@ const FeedbackSubmitPage = () => {
         if (request) {
           if (request.status.toLowerCase() === "submitted" || request.submitDate) {
             setRequestSubmitted(true);
+          } else if (request.recipientId !== currentUserId) {
+            history.push("/checkins");
+            window.snackDispatch({
+              type: UPDATE_TOAST,
+              payload: {
+                severity: "error",
+                toast: "You are not authorized to perform this operation.",
+              },
+            });
           } else {
-            setFeedbackRequest(request);
+              setFeedbackRequest(request);
```

Comment:
This will allow an unauthorized user to access the page if the feedback request has already been submitted by someone else. The check for `request.recipientId !== currentUserId` should come before the check for whether the request was submitted.